File: edge-plugin/src/main/scala/com/cloudentity/edge/plugin/ConvertOpenApiService.java
package com.cloudentity.edge.plugin;

import com.cloudentity.edge.plugin.openapi.*;
import com.cloudentity.tools.vertx.bus.VertxEndpoint;
import com.cloudentity.tools.vertx.tracing.TracingContext;
import io.vertx.core.Future;

public interface ConvertOpenApiService {
  @VertxEndpoint(address = ":plugin.convert-openapi")
  Future<ConvertOpenApiResponse> convertOpenApi(TracingContext ctx, ConvertOpenApiRequest req);
}

File: edge-plugin/src/main/scala/com/cloudentity/edge/plugin/ConvertOpenApiService.scala
package com.cloudentity.edge.plugin

import com.cloudentity.edge.plugin.openapi._
import com.cloudentity.tools.vertx.bus.VertxEndpoint
import com.cloudentity.tools.vertx.tracing.TracingContext
import io.vertx.core.Future

trait ConvertOpenApiService {
  @VertxEndpoint(address = ":plugin.convert-openapi")
  def convertOpenApi(ctx: TracingContext, req: ConvertOpenApiRequest): Future[ConvertOpenApiResponse]
}
